### PR TITLE
Gatekeeper default templates

### LINKED ISSUE #697: Improve OPA+Gatekeeper Documentation

### PR DESCRIPTION
**Pull request by BrendanGalloway**
### Description

Add a set of default constraint templates that are installed when gatekeeper is installed
Enhance gatekeeper docs

### Dependencies

No

### Breaking Change

- [ ] Yes
- [ X ] No

### Testing Notes

**What I did:**
Created a kind test cluster with the karina test script and SUITE='gatekeeper'
Verified that the constrainttemplates are reported as installed during provisioning:
```INFO[0003] constrainttemplates//k8sallowedrepos configured
INFO[0003] constrainttemplates//k8sblocknodeport configured
INFO[0004] constrainttemplates//k8scontainerlimits configured
INFO[0004] constrainttemplates//k8scontainerratios configured
INFO[0004] constrainttemplates//k8sdenyall configured
INFO[0005] constrainttemplates//k8suniqueingresshost configured
INFO[0005] constrainttemplates//k8sdisallowedtags configured
INFO[0006] constrainttemplates//k8srequiredannotations configured
INFO[0006] constrainttemplates//k8srequiredlabels configured
INFO[0006] constrainttemplates//k8srequiredprobes configured
```
checked that constrainttemplates are correctly reported after provisioning:
```
brendan@blap ~$ kubectl get constrainttemplates
NAME                     AGE
k8sallowedrepos          24m
k8sblocknodeport         24m
k8scontainerlimits       24m
k8scontainerratios       24m
k8sdenyall               24m
k8sdisallowedtags        24m
k8srequiredannotations   24m
k8srequiredlabels        24m
k8srequiredprobes        24m
k8suniqueingresshost     24m
```
**How you can replicate my testing:**
Repeat the above (remember to make pack)

### **Links**
Fixes #697

### CODE REVIEW

**Comment by moshloop** [comment]
Can you check the test failures?

File: docs/admin-guide/opa.md [hunk 1]
```diff
@@ -1,15 +1,60 @@
...
+    - unpolicied-namespace
 ```
 
+---
+**NOTE**
+
+Support for installing vanilla OPA via karina is deprecated.  Use at own risk.
+
+---
+
```
Comment (moshloop) [inline docs/admin-guide/opa.md:19]:
```suggestion
!!! warn Deprecated
    Vanilla [OPA with kube-mgmt](https://www.openpolicyagent.org/docs/kubernetes-admission-control.html) can still be deployed using the [opa](/reference/config/#opa), it is however no longer recommended or supported.
```

File: docs/admin-guide/opa.md [hunk 2]
```diff
@@ -1,15 +1,60 @@
...
+#OPA Gatekeeper
 
 ```yaml
-
+gatekeeper:
+  version: v3.3.0
+  constraints: /path/to/constraints/folder
+  templates: /path/to/templates/folder
+  whitelistNamespaces:
+    - unpolicied-namespace
```
Comment (moshloop) [inline docs/admin-guide/opa.md:10]: I think move these out to another section titled, `Adding new constraints`, `Excluding namespaces` with examples of the contents . e.g.

File: docs/admin-guide/opa.md [hunk 3]
```diff
@@ -1,15 +1,60 @@
 `karina.yml`
+#OPA Gatekeeper
```
Comment (moshloop) [inline docs/admin-guide/opa.md:2]:
```suggestion
## Deploying
```

File: docs/admin-guide/opa.md [hunk 4]
```diff
@@ -1,15 +1,60 @@
...
+- K8sDenyAll
+- K8sDisallowedTags
+- K8sRequiredAnnotations
+- K8sRequiredLabels
+- K8sRequiredProbes
+- K8sUniqueIngressHost
+
+Additional templates and constraints can be referred to in the deployment configuration.
+
+To test new constraints, configure them with
```
Comment (moshloop) [inline docs/admin-guide/opa.md:44]: I think we need an example of how to turn on a constraintt

File: docs/admin-guide/opa.md [hunk 5]
```diff
@@ -1,15 +1,60 @@
...
+
+```yaml
+spec:
+  enforcementaction: dryrun
+```
```
Comment (moshloop) [inline docs/admin-guide/opa.md:49]: Fully qualify this
```suggestion
```yaml 
apiVersion: constraints.gatekeeper.sh/v1beta1
kind: K8sAllowedRepos
spec:
  enforcementaction: dryrun
  match: ...
  parameters: ...

```
```

File: docs/admin-guide/opa.md [hunk 6]
```diff
@@ -1,15 +1,60 @@
...
+
+and monitor the number of violations that would be enforced using:
+
+```bash
+# high level overview
+karina status violations -c gatekeeper.yaml
+# specific violation
+kubectl describe constraint $CONSTRAINT
```
Comment (moshloop) [inline docs/admin-guide/opa.md:57]:
```suggestion
kubectl describe constraint K8sAllowedRepos
```

File: docs/admin-guide/opa.md [hunk 7]
```diff
@@ -1,15 +1,60 @@
...
+```
 
-See https://open-policy-agent.github.io/gatekeeper/website/docs/howto/
+https://play.openpolicyagent.org/ is a useful tool for debugging rego operation
```
Comment (moshloop) [inline docs/admin-guide/opa.md:60]:
```suggestion
See [:octicons-link-external-24:  Rego Playground](https://play.openpolicyagent.org) for 
```

File: test/gatekeeper.yaml [hunk 8]
```diff
@@ -0,0 +1,73 @@
+# in order to connect to cluster after provisioning
```
Comment (moshloop) [inline test/gatekeeper.yaml:1]: Shouldn't this be part of the platform test suite ?

Comment (BrendanGalloway) [reply test/gatekeeper.yaml:1]: I didn't realise it was already included there, will remove

**Comment by moshloop** [comment]
@BrendanGalloway please squash

**Comment by moshloop** [comment]
Please check the failing platform test.

**Pull request merged**